langchain4j/src/main/java/dev/langchain4j/service/AiServiceStreamingResponseHandler.java

```diff
@@ -295,12 +299,13 @@ public void onCompleteResponse(ChatResponse chatResponse) {
                 return;
             }
 
-            ChatRequest chatRequest = ChatRequest.builder()
+            ChatRequest nextChatRequest = ChatRequest.builder()
                     .messages(messagesToSend(invocationContext.chatMemoryId()))
                     .toolSpecifications(toolSpecifications)
                     .build();
 
             var handler = new AiServiceStreamingResponseHandler(
+                    chatRequest,
```

Comment:
```suggestion
                    nextChatRequest,
```
Shouldn't it be nextChatRequest?

Reply:
Otherwise ChatRequest and ChatResponse inside the event will not correspond

Reply:
If you need to have an **_initial_** `ChatRequest` inside the event as well, then add a second `ChatRequest initialChatRequest` field

Reply:
in case of the first LLM call `chatRequest == initialChatRequest`, for all next calls in scope of a single AI Service invocation `chatRequest != initialChatRequest`

Reply:
oh - good catch! Thank you. I fixed it.